### PR TITLE
Less requests

### PR DESCRIPTION
**Pull request by MarquiseRosier**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [x] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues
#325 

Thanks for contributing!

### CODE REVIEW

**Review by trieloff — APPROVED** [review]
This works, but you could simplify the code a bit by changing the order to `hlxEmbed, serviceEmbed` so that you could do something like:

```js
return serviceEmbed && hlxEmbed.headers['X-Provider'] !== 'Helix' && serviceEmbed.status !== 400 ? serviceEmbed : hlxEmbed
```

File: src/index.js [hunk 1]
```diff
@@ -137,13 +146,30 @@ async function run(params) {
...
-  if ((params.api || params.OEMBED_RESOLVER_URI) && result.headers['X-Provider'] !== 'Helix') {
-    // filter all __ow_something parameters out
-    return serviceembed(params, urlString, log);
+  // check that OEMBED_RESOLVER_URI and a OEMBED service key are included
+  if ((params.api || params.OEMBED_RESOLVER_URI)) {
+    // add service based embed to be concurrently resolved
+    promises.push(serviceembed(params, urlString, log));
   }
-  return result;
+  promises.push(embed(urlString, params));
```
Comment (tripodsan) [inline src/index.js:155]: if you push this first, you know the 2nd might be optional. this makes the test below easier...

File: src/index.js [hunk 2]
```diff
@@ -137,13 +146,30 @@ async function run(params) {
...
+  const resolvedPromises = await Promise.all(promises);
+
+  // if params for external embed service are provided; the array
+  // of resolved promises will be of length 2 so we test for this case
+  if (promises.length === 2) {
+    [serviceEmbed, hlxEmbed] = resolvedPromises;
+    if (hlxEmbed.headers['X-Provider'] !== 'Helix' && serviceEmbed.status !== 400) {
+      return serviceEmbed;
+    } else {
+      return hlxEmbed;
```
Comment (tripodsan) [inline src/index.js:165]: doesn't lint complain here?

File: src/index.js [hunk 3]
```diff
@@ -137,13 +146,30 @@ async function run(params) {
...
+    // add service based embed to be concurrently resolved
+    promises.push(serviceembed(params, urlString, log));
   }
-  return result;
+  promises.push(embed(urlString, params));
+  const resolvedPromises = await Promise.all(promises);
+
+  // if params for external embed service are provided; the array
+  // of resolved promises will be of length 2 so we test for this case
+  if (promises.length === 2) {
```
[the remark this answers is not in the document]
Comment (MarquiseRosier) [reply src/index.js:160]: Ahhhh nice idea; I absolutely agree here; thanks for this @tripodsan

**Review by tripodsan — APPROVED** [review]

**Pull request merged**